### ISSUE TITLE
Merging pages possible but not working

### ISSUE THREAD

**Issue by ddauber** [athensresearch/athens#552]
*If you aren't sure about your bug, you can always join our [Discord](https://discord.gg/HNmxvpm) and ask in #agora or #engineering!*

**Problem**

When creating the same page, Athens is aware of it and asks to merge the pages. Once confirmed it throws an error message (see screenshot below)

**Expected Behavior**

Two pages merge and content created on both pages is merged based on creation date (chronologically on block-level) or the entire content of the younger page is appended to the older page at the bottom.

**Dependencies**

Operating system: Macbook Pro M1, Big Sur Version 11.1 

- Athens: Beta.26 (for Mac)
- browser: 
- JVM:
- node.js: 

**To Reproduce**

Create two pages with the same name.

**Screenshots**

<img width="285" alt="Screenshot 2021-01-19 at 11 17 38" src="https://user-images.githubusercontent.com/62331069/105030425-ae678900-5a4b-11eb-8a19-fad17440e7dd.png">

**Comment by tangjeff0** [comment]
Hi @ddauber can you restart Athens, triggering an auto-update, and test this out with version beta.27?

**Comment by ddauber** [comment]
I tried it now under Beta.27

I created to pages: **test merge** and **test merge 2**

I added text to both of them, different from each other.

Scenario 1: Changing the page name of **test merge 2** to **test merge** on the page itself
Result: Athens freezes -> restart -> **test merge 2** is back (so changes did not get saved) -> page works as usual (and can be deleted, etc.

Scenario 2: Changing the page name of **test merge 2** to **test merge** while test merge 2 is in the right-side panel
Result: Warning that page exists and suggests that I can merge it. UI is a bit broken here, but I just increased the window size to ensure I can click the 'accept' icon. This worked, although Athens crashed a couple of times.

I repeated Scenario 2 with another set of pages and I got this error message. I created the 'test' page, then created a new page from the 'test' page by making a page reference '[[test2]]. I then opened that page in the right-hand pane and it returned the below error message when I tried to change the page name and have them merged. However, Athens did not crash. I managed to continue to edit the pages.

<img width="281" alt="Screenshot 2021-01-19 at 20 09 20" src="https://user-images.githubusercontent.com/62331069/105087352-3e302600-5a92-11eb-9ff2-a2daad0ec30f.png">

I then remove the page reference from 'test' (i.e. remove [[test2]]) and it worked and merged my content of both pages.

So, maybe one issue lies in how block and page references are handled when merging. It would result in a self-reference? I noticed from the 'Welcome' page that self-referential links aren't an issue.

Let me know if you need more insights. Also happy to record a short video to show the problem I encountered if the explanations do not suffice.

Best wishes

**Comment by tangjeff0** [comment]
Hi Daniel, the issue has to do with datascript. So sorry for this. Just pushed beta.28, which deletes orphan pages less aggressively. Let me know if you continue to have problems. @ddauber

**Comment by ddauber** [comment]
I am afraid nothing has changed. I can confirm that with all kinds of different combinations, the result remains the same: Merging pages which are referenced elsewhere cannot be merged until those references have been removed. I guess a quick workaround could be to 'Unlink All', i.e. the opposite feature you recently implemented. This would make merging possible, but it would, of course, retain the unlinked references to a page that does not exist. This would have to be manually adjusted.

In short: I am sure this is not what you have in mind. :)

**Comment by tangjeff0** [comment]
Hey @ddauber, you're right, this still doesn't work. I'll have time to address this in ~3 hours. Thanks so much for your patience.

**Comment by ddauber** [comment]
No hurry. Happy if I manage to help a bit.

**Comment by tangjeff0** [comment]
Hi @ddauber, this should be fixed in the latest version beta.29! Thanks so much for your help with this.

**Comment by ddauber** [comment]
Hi @tangjeff0 

great. Thanks. My Athens has not updated just yet, but I look forward to Beta.29.

**Comment by tangjeff0** [comment]
It's updated when you restart, FYI @ddauber

**Comment by ddauber** [comment]
Yes, I did this successfully for Beta.26-28, but Beta.29 would not come through for me. I even went back to the download link you provided (Beta.25) and reinstalled it.

I am running out of ideas why it suddenly would stop updating. Is Beta.29 on Windows, but not on Mac yet? @tangjeff0

**Comment by tangjeff0** [comment]
@ddauber sorry, I didn't tag the latest release, so it didn't deploy! Need to start auto-mating this...

**Comment by ddauber** [comment]
Thanks, @tangjeff0 . I truly admire your and your entire team's commitment. Amazing to have such a great tool.

Keep up the good work. Happy to help where I can, although I don't know Clojure (just yet). :)

**Comment by tangjeff0** [comment]
Thanks @ddauber. `beta.29` introduced a tiny bug on startup, just fixed. .30 should be good now!
